docs/source/components/configuration.rst

```diff
@@ -420,6 +422,29 @@ Enables the extraction of marked RST text from source code comments. When enable
    [codelinks.projects.my_project.analyse]
    get_rst = false
 
+.. _`git_root`:
+
+git_root
+^^^^^^^^
+
+Specifies an explicit path to the Git repository root directory. This option is particularly useful in environments where the standard Git root auto-detection fails, such as:
+
+- **Bazel builds**: Where the execution path differs from the standard repository layout
+- **Deeply nested configurations**: Where ``conf.py`` is located in a deep subdirectory far from the repository root
+- **Custom build systems**: Where the working directory is different from the source repository
+
+When not set, **Sphinx-CodeLinks** will automatically traverse parent directories to locate the ``.git`` folder.
+
+**Type:** ``str`` (path)
+**Default:** Not set (auto-detection)
+
+.. code-block:: toml
+
+   [codelinks.projects.my_project.analyse]
+   git_root = "/absolute/path/to/repo"
```

Comment:
Could it be relative to our TOML configuration file like what `src_dir` works? I feel it's optional and can be adapted later if requested